core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -320,14 +321,35 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
         .filter { entry =>
           try {
             val prevFileSize = fileToAppInfo.get(entry.getPath()).map{_.fileSize}.getOrElse(0L)
+
+            def canAccess = {
+              val perm = entry.getPermission
+              val ugi = UserGroupInformation.getCurrentUser
+              val user = ugi.getShortUserName
+              val groups = ugi.getGroupNames
+              if (user == entry.getOwner && perm.getUserAction.implies(FsAction.READ)) {
+                true
+              } else if (groups.contains(entry.getGroup) &&
+                  perm.getGroupAction.implies(FsAction.READ)) {
+                true
+              } else if (perm.getOtherAction.implies(FsAction.READ)) {
+                true
+              } else {
+                throw new AccessControlException(s"Permission denied: user=$user, " +
```

Comment:
Why throw an exception instead of returning false? That makes the function's interface weird.

You could just log the access issue here if you want, and then you can even remove the `try..catch`.